libp2p/crypto/minasn1.nim

```diff
@@ -519,15 +520,15 @@ proc read*(ab: var Asn1Buffer): Asn1Result[Asn1Field] =
         if length != 1:
           return err(Asn1Error.Incorrect)
 
-        if not ab.isEnough(int(length)):
+        if not ab.isEnough(cast[int](length)): # FIXME uint64 to int potentially unsafe
```

Comment:
In these cases we shouldn't cast since this would fail silently by producing junk value

Reply:
what should be done?

Reply:
In this case, I think we should return a int64 from `getLength` (and an error if it overflows), that would avoid all of those

Reply:
Please review the new code